src/client.ts

```diff
@@ -5888,6 +5899,9 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
         // There can be only room-kind push rule per room
         // and its id is the room id.
         if (this.pushRules) {
+            if (!this.pushRules[scope] || !this.pushRules[scope].room) {
+                return;
+            }
```

Comment:
Defensive programming when the server returns `{}`. There's no guarantees that `scope` will exist.